src/Couchbase.Lite.Shared/Query/XQuery.cs

```diff
@@ -120,7 +120,7 @@ public override unsafe IResultSet Execute()
                 }
 
                 Check();
```

Comment:
Is this because Check() guarantees not null?  If so, could Check() return the non-null query?

Reply:
I had tried to do that previously but it ended up being too awkward for reasons I can't remember.  I should try again at some point.